spikes/pikepdf-write/write_tagged_pdf.py

```
@@ -122,29 +137,74 @@ def write_tagged_pdf(
+            #    actually references).
+            if not assignments:
+                continue
+
+            max_mcid = -1
+            for mcid, zone_idx in assignments:
+                if 0 <= zone_idx < len(zone_elems):
+                    zone_elems[zone_idx].K.append(mcid)
+                    max_mcid = max(max_mcid, mcid)
+
+            mcid_to_elem = pikepdf.Array()
+            for mcid in range(max_mcid + 1):
```

Ruff (0.15.12)

[warning] Loop control variable `mcid` not used within loop body

(B007)